manifests/repo/debian.pp

```diff
@@ -62,4 +62,35 @@
       }
     }
   }
+
+  if ($php::globals::php_version == '7.1') {
+    # Required packages for PHP 7.1 repository
+    package { 'apt-transport-https':
```

Comment:
I do see that this package is required, but this could break others configuration if they defined the package in a profile. 

the same for the other packages declared here.

I really would not do business of adding repos. and such packags. If doing the roles/profiles pattern it should be solved in the profile level. For me this module should not try to add repos. but thats my personal opinion. so many things fail if you would run such things in a restricted environment (no internet access, and the policy to use internal repos).

Right now I can't say what to do with these packages. :-/